services/bettersocial/post/createPostV3.js

```diff
@@ -0,0 +1,320 @@
+const {
+  POST_TYPE_STANDARD,
+  POST_TYPE_POLL,
+  POST_VERB_POLL,
+  POST_VERSION,
+} = require('../../../helpers/constants');
+const { addForCreatePost } = require('../../score');
+const PostFunction = require('../../../databases/functions/post');
+const PollingFunction = require('../../../databases/functions/polling');
+const PostAnonUserInfoFunction = require('../../../databases/functions/postAnonUserInfo');
+const { convertLocationFromModel } = require('../../../utils');
+const ErrorResponse = require('../../../utils/response/ErrorResponse');
+
+const isEmptyMessageAllowed = (body) => {
+  const isPollPost = body?.verb === POST_VERB_POLL;
+  const isBodyEmpty = !body?.message || body?.message === '';
+  const isImageUrlEmpty = !body?.images_url || body?.images_url.length === 0;
+
+  return isBodyEmpty && isImageUrlEmpty && !isPollPost;
+};
+
+/**
+ *
+ * @param {import('express').Request} req
+ * @param {boolean} isAnonimous
+ * @returns
+ */
+const BetterSocialCreatePostV3 = async (req, isAnonimous = true) => {
+  const { userId, body } = req;
+  const filteredTopics = filterAllTopics(body?.message, body?.topics);
+
+  let userDetail = {};
+  let data = {};
+  let locationDetail = {};
+  let locationToPost = 'Everywhere';
+  let locationTO = null;
+  let post = {};
+  let uploadedImages = body?.images_url || [];
+
+  if (isEmptyMessageAllowed(req?.body))
+    return {
+      success: false,
+      message: 'Post cannot be empty',
+    };
+
+  try {
+    /**
+     * Base Post Process
+     */
+    if (isAnonimous)
+      userDetail = await UsersFunction.findAnonymousUserId(User, userId);
+    else userDetail = await UsersFunction.findUserById(User, userId);
+    const getstreamObjectParam = generateDefaultGetstreamObject(
+      body,
+      isAnonimous,
+      userDetail
+    );
+    console.log({ userDetail });
+    if (!body?.is_photo_uploaded)
+      uploadedImages = await CloudinaryService.uploadBase64Array(
+        body?.images_url
+      );
+
+    const feedExpiredAt = getFeedDuration(body?.duration_feed);
+    locationDetail = await LocationFunction.getLocationDetail(
+      Locations,
+      body?.location_id
+    );
+
+    if (body?.location !== 'Everywhere') {
+      locationToPost = convertLocationFromModel(locationDetail);
+      locationTO = convertLocationFromModel(locationDetail, true);
+    }
+
+    data = {
+      verb: body?.verb,
+      message: body?.message,
+      topics: filteredTopics,
+      privacy: body?.privacy,
+      object: getstreamObjectParam,
+      anonimity: isAnonimous,
+      location: locationToPost,
+      duration_feed: body?.duration_feed,
+      images_url: uploadedImages,
+      expired_at: feedExpiredAt,
+      count_upvote: 0,
+      count_downvote: 0,
+      post_type: POST_TYPE_STANDARD,
+      version: POST_VERSION,
+      to: handleCreatePostTO(
+        userDetail.user_id,
+        req?.body,
+        isAnonimous,
+        locationTO
```

Comment:
mas @falhar on handleCreatePostTO, you cannot change this to userDetail.user_id
Please check the function, signedUserId is still needed to push feed to main_feed:signedUserId.

I suggest you add more parameter to the function let's say targetNotificationUserId, and check this
if it is set push notification:targetNotificationUserId
if it is null push notification:signedUserId

`
const handleCreatePostTO = (userId, postBody, isAnonimous = true, locationString = null, targetNotificationUserId = null) => {
...
if(targetNotificationUserId) TO.push(notification:targetNotificationUserId
else TO.push(userId)
...
}
`